module/FinnaSearch/src/FinnaSearch/Backend/L1/Connector.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * L1 connector.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+namespace FinnaSearch\Backend\L1;
+
+use VuFindSearch\ParamBag;
+
+/**
+ * L1 connector.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class Connector extends \VuFindSearch\Backend\Solr\Connector
+{
+    /**
+     * Send query to SOLR and return response body.
+     *
+     * @param string   $handler SOLR request handler to use
+     * @param ParamBag $params  Request parameters
+     *
+     * @return string Response body
+     */
+    public function query($handler, ParamBag $params)
+    {
+        // Restrict all queries to AOE records only.
+        $params->add('fq', 'datasource_str_mv:aoe');
```

Comment:
Halutaanko varmasti tehdä näin kovakoodattuna?

Reply:
En tiedä... 😊 Pitäisikö laittaa `const`:iksi vai ihan asetukseksi?

Reply:
Normaalisti käytettävät asetukset on jo olemassa (HiddenFilters tai RawHiddenFilters). Jos halutaan kuitenkin jotenkin pakottaa jotain filtteriksi, niin vaikka member-muuttujaksi, jolla on nykyinen filtteri oletusarvona, mutta jonka voisi tarvittaessa yliajaa asetuksella. Asetus olisi valmiina ini-tiedostossa kommentoituna pois käytöstä. Näin ei tarvita softamuutosta, jos jotain muuttuu, esim. aoe tuleekin useampana settinä.

Reply:
Mihin ini-tiedostoon ja osioon asetus olisi paras laittaa? `L1.ini` ja oma osio `[L1Filters]`?

Reply:
L1.ini joo. Jos sitä ei haluta HiddenFilters- tai RawHiddenFilters-sektioon, niin sitten ehkä BackendFilters? Ennemmin jotain ei-L1-spesifistä, niin voidaan käyttää samaa, jos vaikka tulee jokin toinen vastaava backend myöhemmin.

Reply:
Mikä olisi paras tapa antaa konfiguraatio `Connector`-luokalle? Lisäparametrina `__construct()`:ille, jolloin instanssilla on varmasti aina asetukset kunnossa, mutta aiheuttaa lisätoimenpiteitä jos yläjuoksukin päättää lisätä uuden parametrin. Vai erillisellä `setConfig()`-tyyppisellä metodilla heti instanssin luonnin (`parent::createConnector()`-kutsun) jälkeen?

Reply:
Hmm, vai lisätäänkö `AbstractSolrBackendFactory::createConnector()`:sissa `$handlers`:iin?

Reply:
ks. SolrDefaultBackendFactory::getHiddenFilters.